File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.rb
# @param {Integer[]} nums
# @return {Integer}
def find_min(nums)
  low = 0
  high = nums.length - 1
  result = nums[0]
  while low < high
    return nums[low] if nums[low] <= nums[high]
    mid = (low + high) / 2
    if nums[mid] >= nums[low]
      low = mid + 1
    else
      high = mid
    end
  end
  return nums[low]
end

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMin = function(nums) {
    let left = 0, right = nums.length -1;
    
    while(left < right){
        if(nums[left] < nums[right])
            return nums[left];
        
        let mid = Math.floor((left+right)/2);
        if(nums[mid] >= nums[left]){
            left = mid + 1;
        }
        else{
            right = mid;
        }
    }
    return nums[left];
};